ui/src/app/views/rename-modal/component.tsx

```diff
@@ -0,0 +1,73 @@
+import * as React from 'react';
+
+import {
+  Button
+} from 'app/components/buttons';
+import {
+  Modal,
+  ModalBody,
+  ModalFooter,
+  ModalTitle,
+} from 'app/components/modals';
+import {ReactComponent} from 'app/utils';
+
+
+interface RenameModalProps {
+  resource: any;
+  resourceType: string;
+  onRename: Function;
+  onCancel: Function;
+}
+interface RenameModalState {
+  loading: boolean;
+  newName: string;
+}
+
+@ReactComponent({
+  selector: 'app-rename-modal',
+  propNames: ['resource', 'resourceType', 'onRename', 'onCancel']
+})
+export class RenameModalComponent extends React.Component<RenameModalProps, RenameModalState> {
+  constructor(props: RenameModalProps) {
+    super(props);
+    this.state = {
+      loading: false,
+      newName: ''
+    };
+  }
+
+  private emitRename(): void {
+    if (this.state.loading) {
```

Comment:
nit: This check should be unnecessary, since the button's `disabled` prop will prevent this from being called.

Reply:
I wasn't sure about a possible race here, but should probably be fine - done.